### DISCUSSION TITLE
How many devices sending notifications can be handled using Bleak?

### DISCUSSION THREAD

**Discussion by SergioEanX** [hbldh/bleak#1302] [Q&A] [answered]
I'm interested to develop a Python application to be run on a Raspberry 3/4 to receive notifications from 15 or more devices concurrently.
In the examples folder, there is a useful template for [two devices](https://github.com/hbldh/bleak/blob/develop/examples/two_devices.py), but I wonder how many devices can be effectively handled in this way.
Any experience/suggestion in this regard?
To use Go or Rust could be a better alternative or everything holds the same as, I suppose, most  BLE modules/packages use [BlueZ](http://www.bluez.org/) via the [D-Bus](https://en.wikipedia.org/wiki/D-Bus) interface?
Thanks!

**Comment by dlech** [comment, marked as answer]
The number of concurrent connections will be limited by what your Bluetooth adapter is capable of. I don't think I've seen a Bluetooth chip capable of more than 8 (but I been looking either).
